### PR TITLE
[PDCL-8704] Add Weather Schema and FieldGroups

### PR DESCRIPTION
**Pull request by alinaeftn**
Please link to the issue #…

### CODE REVIEW

**Review by mflorin — APPROVED** [review]

File: components/classes/weather.class.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (mjasrotia) [inline components/classes/weather.class.json:1]: please rename this to weather.schema.json

Comment (mjasrotia) [inline components/classes/weather.class.json:1]: Please rename this the file to weather.schema.json

File: components/fieldgroups/weather/currentWeather.fieldgroup.json [hunk 2]
```diff
@@ -0,0 +1,257 @@
+{
```
Comment (mjasrotia) [inline components/fieldgroups/weather/currentWeather.fieldgroup.json:1]: Please rename this the file to "current-weather.schema.json". Also move it under shared folder (fieldgroups/shared)

File: components/fieldgroups/weather/currentWeather.fieldgroup.json [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+    "type": "object",
+    "description": "Current weather conditions for a postal code.",
+    "definitions": {
+        "customFields": {
+            "properties": {
+                "xdm:weather": {
+                    "type": "object",
+                    "properties": {
+                        "xdm:current": {
+                            "title": "current",
```
Comment (mjasrotia) [inline components/fieldgroups/weather/currentWeather.fieldgroup.json:26]: current ==> Current

File: components/fieldgroups/weather/currentWeather.fieldgroup.json [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+                                        "11": "Extreme",
+                                        "12": "Extreme",
+                                        "13": "Extreme",
+                                        "14": "Extreme",
+                                        "15": "Extreme",
+                                        "16": "Extreme",
+                                        "-2": "Not Available",
+                                        "-1": "No Report"
+                                    },
+                                    "meta:xdmType": "string"
```
Comment (mjasrotia) [inline components/fieldgroups/weather/currentWeather.fieldgroup.json:151]: We can remove this

File: components/fieldgroups/weather/forecasted-weather.fieldgroup.json [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+        "https://ns.adobe.com/xdm/context/experienceevent",
+        "https://ns.adobe.com/xdm/classes/weather"
+    ],
+    "meta:status": "experimental",
+    "title": "Forecasted Weather",
+    "type": "object",
+    "description": "Forecasted weather conditions for a postal code",
+    "definitions": {
+        "customFields": {
+            "type": "object",
```
Comment (mjasrotia) [inline components/fieldgroups/weather/forecasted-weather.fieldgroup.json:21]: remove this type

File: components/fieldgroups/weather/forecasted-weather.fieldgroup.json [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+            "properties": {
+                "xdm:weather": {
```
Comment (mjasrotia) [inline components/fieldgroups/weather/forecasted-weather.fieldgroup.json:23]: Add "type": "object" here

File: components/fieldgroups/weather/forecasted-weather.fieldgroup.json [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+                    "properties": {
+                        "xdm:forecast": {
+                            "title": "Forecast",
+                            "description": "Forecasted weather at the time the event was generated.",
+                            "isRequired": false,
```
Comment (mjasrotia) [inline components/fieldgroups/weather/forecasted-weather.fieldgroup.json:28]: no need for this. Please remove

File: components/fieldgroups/weather/forecasted-weather.fieldgroup.json [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+    "type": "object",
+    "description": "Forecasted weather conditions for a postal code",
+    "definitions": {
+        "customFields": {
+            "type": "object",
+            "properties": {
+                "xdm:weather": {
+                    "properties": {
+                        "xdm:forecast": {
+                            "title": "Forecast",
```
Comment (mjasrotia) [inline components/fieldgroups/weather/forecasted-weather.fieldgroup.json:26]: Add "type": "object"

File: components/fieldgroups/weather/forecasted-weather.fieldgroup.json [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+                            "description": "Forecasted weather at the time the event was generated.",
+                            "isRequired": false,
+                            "properties": {
+                                "xdm:expiration": {
+                                    "format": "date-time",
+                                    "type": "string",
+                                    "description": "When the fields should no longer be populated.",
+                                    "title": "expiration"
```
Comment (mjasrotia) [inline components/fieldgroups/weather/forecasted-weather.fieldgroup.json:34]: e ==> E

File: components/fieldgroups/weather/forecasted-weather.fieldgroup.json [hunk 10]
```diff
@@ -0,0 +1,343 @@
...
+        },
+        "longRange": {
+            "properties": {
+                "xdm:qpf": {
+                    "title": "qpf",
+                    "description": "The forecasted measurable precipitation (liquid or liquid equivalent) during the 24 hour period.",
+                    "$ref": "https://ns.adobe.com/xdm/mixins/current-weather#/definitions/precipitation"
+                },
+                "xdm:calendarDayTemperatureMax": {
+                    "title": "calendarDayTemperatureMax",
```
Comment (mjasrotia) [inline components/fieldgroups/weather/forecasted-weather.fieldgroup.json:120]: Let's make the title a proper sentence. "Calendar Day Temperature Max". Please apply this in every other title

File: components/fieldgroups/weather/forecasted-weather.fieldgroup.json [hunk 11]
```diff
@@ -0,0 +1,343 @@
...
+                        "11": "Extreme",
+                        "12": "Extreme",
+                        "13": "Extreme",
+                        "14": "Extreme",
+                        "15": "Extreme",
+                        "16": "Extreme",
+                        "-2": "Not Available",
+                        "-1": "No Report"
+                    },
+                    "meta:xdmType": "string"
```
Comment (mjasrotia) [inline components/fieldgroups/weather/forecasted-weather.fieldgroup.json:195]: Please remove this

File: components/fieldgroups/weather/forecasted-weather.fieldgroup.json [hunk 12]
```diff
@@ -0,0 +1,343 @@
...
+                    "description": "Weather information for a nighttime period",
+                    "$ref": "#/definitions/dayPart"
+                },
+                "xdm:qpfSnow": {
+                    "title": "qpfSnow",
+                    "description": "The forecasted measurable precipitation as snow during the 12 or 24 hour forecast period. Measured in centimeters.",
+                    "$ref": "https://ns.adobe.com/xdm/mixins/current-weather#/definitions/snowfall"
+                }
+            },
+            "type": "object"
```
Comment (mjasrotia) [inline components/fieldgroups/weather/forecasted-weather.fieldgroup.json:110]: This is just a definition, no need to specify "type": "object" here

File: components/fieldgroups/weather/forecasted-weather.fieldgroup.json [hunk 13]
```diff
@@ -0,0 +1,343 @@
...
+                        "14": "Extreme",
+                        "15": "Extreme",
+                        "16": "Extreme",
+                        "-2": "Not Available",
+                        "-1": "No Report"
+                    },
+                    "meta:xdmType": "string"
+                }
+            },
+            "type": "object"
```
Comment (mjasrotia) [inline components/fieldgroups/weather/forecasted-weather.fieldgroup.json:198]: This is just a definition, no need to specify "type": "object" here

File: components/fieldgroups/weather/forecasted-weather.fieldgroup.json [hunk 14]
```diff
@@ -0,0 +1,343 @@
...
+                "precipChance": {
+                    "title": "precipChance",
+                    "description": "",
+                    "type": "integer"
+                },
+                "precipType": {
+                    "title": "precipType",
+                    "description": "",
+                    "type": "string",
+                    "pattern": "rain|snow|precip"
```
Comment (mjasrotia) [inline components/fieldgroups/weather/forecasted-weather.fieldgroup.json:216]: This should be enum with values ["rain", "snow", "precip"]

**Comment by mjasrotia** [comment]
Consolidated all the changes in the new PR
https://github.com/adobe/xdm/pull/1577

**Pull request closed without merging**